core/src/test/java/org/elasticsearch/action/support/replication/ReplicationResponseTests.java

```diff
@@ -113,6 +115,156 @@ public void testFailureEqualsAndHashcode() {
         checkEqualsAndHashCode(randomFailure(), copy, mutate);
     }
 
+    public void testShardInfoToXContent() throws IOException {
+        ReplicationResponse.ShardInfo shardInfo = new ReplicationResponse.ShardInfo(10, 10);
+
+        try (XContentBuilder builder = XContentBuilder.builder(XContentType.JSON.xContent())) {
+            shardInfo.toXContent(builder, ToXContent.EMPTY_PARAMS);
+            assertEquals("{\"total\":10,\"successful\":10,\"failed\":0}", builder.string());
+        }
+    }
+
+    public void testShardInfoWithFailureToXContent() throws IOException {
+        ReplicationResponse.ShardInfo shardInfo = new ReplicationResponse.ShardInfo(2, 1,
+                new ReplicationResponse.ShardInfo.Failure(new ShardId("_index", "_uuid", 1), "_node_id",
+                        new IllegalStateException("failure cause"), RestStatus.LOCKED, true));
+
+        try (XContentBuilder builder = XContentBuilder.builder(XContentType.JSON.xContent())) {
+            shardInfo.toXContent(builder, ToXContent.EMPTY_PARAMS);
+            assertEquals("{"
+                + "\"total\":2,"
+                + "\"successful\":1,"
+                + "\"failed\":1,"
+                + "\"failures\":["
+                    + "{"
+                    + "\"_index\":\"_index\","
+                    + "\"_shard\":1,"
+                    + "\"_node\":\"_node_id\","
+                    + "\"reason\":{\"type\":\"illegal_state_exception\",\"reason\":\"failure cause\"},"
+                    + "\"status\":\"LOCKED\","
+                    + "\"primary\":true"
+                    + "}"
+                + "]"
+            + "}", builder.string());
+        }
+    }
+
+    public void testRandomShardInfoToXContent() throws IOException {
+        final ReplicationResponse.ShardInfo shardInfo = randomShardInfo();
+
+        // Build the expected JSON from the random ShardInfo
+        StringBuilder expectedJson = new StringBuilder();
+        expectedJson.append("{");
```

Comment:
I see, I was hoping we could also test other formats, we test only json here? using a builder we could test also other formats?